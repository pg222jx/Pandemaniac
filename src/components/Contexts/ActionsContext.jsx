import React,{ useState, createContext } from 'react';

export const ActionsContext = createContext()

export const ActionsProvider = props => {

    const [amount, setAmount] = useState(2)

    return (
        <ActionsContext.Provider value={[amount, setAmount]}>
            {props.children}
        </ActionsContext.Provider>
    )
}

