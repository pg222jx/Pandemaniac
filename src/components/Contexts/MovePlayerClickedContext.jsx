import React,{ useState, createContext } from 'react';

export const MovePlayerClickedContext = createContext()

export const MovePlayerClickedProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <MovePlayerClickedContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </MovePlayerClickedContext.Provider>
    );
}

