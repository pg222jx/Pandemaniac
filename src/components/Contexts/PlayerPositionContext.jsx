import React,{ useState, createContext } from 'react';

export const PlayerPositionContext = createContext()

export const PlayerPositionProvider = props => {

    const [position, setPosition] = useState({bottom: undefined, left: undefined})

    return (
        <PlayerPositionContext.Provider value={[position, setPosition]}>
            {props.children}
        </PlayerPositionContext.Provider>
    )
}

