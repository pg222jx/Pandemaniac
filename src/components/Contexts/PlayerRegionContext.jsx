import React,{ useState, createContext } from 'react';

export const PlayerRegionContext = createContext()

export const PlayerRegionProvider = props => {

    const [region, setRegion] = useState('')

    return (
        <PlayerRegionContext.Provider value={[region, setRegion]}>
            {props.children}
        </PlayerRegionContext.Provider>
    )
}

