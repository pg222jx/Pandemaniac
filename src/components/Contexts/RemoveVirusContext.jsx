import React,{ useState, createContext } from 'react';

export const RemoveVirusContext = createContext()

export const RemoveVirusProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <RemoveVirusContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </RemoveVirusContext.Provider>
    );
}

