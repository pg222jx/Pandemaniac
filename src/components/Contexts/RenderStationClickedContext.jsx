import React,{ useState, createContext } from 'react';

export const RenderStationClickedContext = createContext()

export const RenderStationClickedProvider = props => {

    const [clicked, setClicked] = useState(false)

    return (
        <RenderStationClickedContext.Provider value={[clicked, setClicked]}>
            {props.children}
        </RenderStationClickedContext.Provider>
    );
}

