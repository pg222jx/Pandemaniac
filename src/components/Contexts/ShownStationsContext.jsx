import React, { useState, createContext } from 'react'

export const ShownStationsContext = createContext()

export const ShownStationsProvider = props => {

    const [shownStations, setShownStations] = useState([])

    return (
        <ShownStationsContext.Provider value={[shownStations, setShownStations]}>
            {props.children}
        </ShownStationsContext.Provider>
    )
}

