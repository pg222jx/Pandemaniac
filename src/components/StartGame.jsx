/**
 * Representing the 'Start Game' button.
 *
 * @author Pernilla GÃ¶th
 * @version 1.0.0
 */


import React, { useState, useContext } from 'react'
import { PlayerPositionContext } from './Contexts/PlayerPositionContext'
import { PlayerRegionContext } from './Contexts/PlayerRegionContext'

const StartGame = (props) => {
    const [buttonVisibility, setButtonVisibility] = useState('visible')
    const [position, setPosition] = useContext(PlayerPositionContext)// eslint-disable-line no-unused-vars
    const [playerRegion, setPlayerRegion] = useContext(PlayerRegionContext)// eslint-disable-line no-unused-vars

    /**
     * Starts the game.
     */
    const start = () => {
        props.timer()
        setButtonVisibility('hidden')
        randomizeContinent()
    }
 
     /**
      * Gets a random continent.
      */
    const randomizeContinent = () => { 
         const continents = [1, 2, 3, 4, 5, 6]
         const randomContinent = continents[Math.floor(Math.random() * continents.length)];
         setPlayerPosition(randomContinent)
     }
 
     /**
      * Sets the coordinate for the player according to continent.
      * 
      * @param {number} - Representing a continent.
      */
     const setPlayerPosition = (continent) => {
        if (continent === 1) {
            setPlayerRegion('NA')
            setPosition({bottom: '350px', left: '150px'})
        } else if (continent === 2) {
            setPlayerRegion('SA')
            setPosition({bottom: '110px', left: '235px'})
        } else if (continent === 3) {
            setPlayerRegion('AF')
            setPosition({bottom: '200px', left: '530px'})
        } else if (continent === 4) {
            setPlayerRegion('EU')
            setPosition({bottom: '390px', left: '490px'})
        } else if (continent === 5) {
            setPlayerRegion('AS')
            setPosition({bottom: '365px', left: '700px'})
        } else if (continent === 6) {
            setPlayerRegion('AU')
            setPosition({bottom: '100px', left: '860px'})
        } 
     }

    return (
        <React.Fragment>
        <div className="startButton" style={{visibility: buttonVisibility}} onClick={() => start()}>
            <button>Start Game!</button>
        </div>
        </React.Fragment>
    )
}
 
export default StartGame