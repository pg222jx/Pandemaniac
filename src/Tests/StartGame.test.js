import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { shallow, mount } from 'enzyme'
import { PlayerPositionProvider } from '../components/Contexts/PlayerPositionContext'
import { PlayerRegionProvider } from '../components/Contexts/PlayerRegionContext'
import StartGame from '../components/StartGame'

const wrapperShallow = shallow(
    <PlayerPositionProvider>
        <PlayerRegionProvider>
            <StartGame/>
        </PlayerRegionProvider>
    </PlayerPositionProvider>
)

const wrapperMount = mount(
    <PlayerPositionProvider>
        <PlayerRegionProvider>
            <StartGame />
        </PlayerRegionProvider>
    </PlayerPositionProvider>
)

  describe('Testing the StartGame component', () => {
    it('should match the snapshot', () => {
        expect(wrapperShallow.html()).toMatchSnapshot()
      })
      
    it('should have 1 button', () => {
          expect(wrapperMount.find('button').length).toEqual(1)
    })

    test('that the button is set to hidden after click', async () => {
        const timer = jest.fn()
        
        const { getByText, findByText } = render(
            <PlayerPositionProvider>
                <PlayerRegionProvider>
                    <StartGame timer={() => {timer()}}/>
                </PlayerRegionProvider>
            </PlayerPositionProvider>)

        const button = getByText('Start Game!')
        expect(button).toBeVisible()
        fireEvent.click(button)
        const buttonAfter = await findByText('Start Game!')
        expect(buttonAfter).not.toBeVisible()
    })
  })