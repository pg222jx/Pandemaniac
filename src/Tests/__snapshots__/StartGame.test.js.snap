// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the StartGame component should match the snapshot 1`] = `"<div class=\\"startButton\\" style=\\"visibility:visible\\"><button>Start Game!</button></div>"`;
